.github/workflows/test.yml

```diff
@@ -0,0 +1,43 @@
+name: Lint,Test,Deploy
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+jobs:
+  lint_test_deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Node setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - name: install dependencies
+        run: npm ci
+
+      - name: lint
+        run: npm run lint:affected
+
+      - name: test
+        run:  npm run test:affected
+
+      - name: deploy
+        if: ${{ github.event.name == 'push'}}
```

Comment:
Here, we need a more advanced pattern. The branch name should be prefixed with the name of the environment.

`<env>-branch-name` (e.g, `stg-my-feature`, `prod-my-feature`)

Whenever we open a pull request, we should use it to deploy it to the right environment. For example, `stg-my-feature` should set the `NODE_ENV` to `stg`

Reply:
A push to master should set `NODE_ENV` to `prod`.

Reply:
I think that we can use a filter, for example:

```
- name: deploy
if: contains(github.ref, "stg")
env:
     NODE_ENV: stg
run:  npm run deploy:affected
 ```

Reply:
@NetanelBasal  have made changes to the workflow pls kindly review

Reply:
Thanks.

- Did you verify that it works? 
- Can you add a new section in the README  that explains the behavior, please?

Reply:
![image](https://user-images.githubusercontent.com/82381308/169844029-92804498-b353-4c37-8117-b9b964255072.jpeg)

I don't know much about nx but I'm getting this error message

Reply:
https://stackoverflow.com/questions/67200882/nx-monorepo-fatal-no-such-ref-main1

Reply:
i have also updated README that explains the behavior of ci/cd @NetanelBasal kindly check it out

Reply:
Why is the deploy isn't with a checkmark?

Reply:
because the branch name must be prefixed with the name of the environment and also deploy it to the right environment

Reply:
Can you please test it in real with your AWS env? I want to make sure that it works.

Reply:
You can push it to your own repo and test it

Reply:
![image](https://user-images.githubusercontent.com/82381308/169977365-53ea998d-3ab0-41d0-8474-4e4ca4673bfa.jpeg)

Deploy to prod is getting this error message

Reply:
Try `nx affected --target=deploy`

Reply:
![image](https://user-images.githubusercontent.com/82381308/169982494-adb4fc21-ecb6-431c-bb79-aae387be8f4d.jpeg)

Reply:
The command works 
Guess it's my Aws config that made it fail

Reply:
Yes, you should update prod.enviroment.ts

Reply:
Did it deploy to your AWS account?

Reply:
🔥